### DISCUSSION TITLE
Prisma relation with same field

### DISCUSSION THREAD

**Discussion by ArthurRF** [prisma/prisma#10457] [Q&A]
Hey there!

I'm having a difficult to use the prisma schema in a specific situation.
The simplified models:

```prisma
model Challenge {
  id              String           @id @default(uuid()) @db.Uuid
  user_challenges UserChallenges[]
  winners         UserChallenges[] @relation("ChallengeWinners")
  
  @@map("challenges")
}

model UserChallenges {
  challenge_id String     @db.Uuid
  user_id      String     @db.Uuid
  challenge    Challenge  @relation(fields: [challenge_id], references: [id])
  winners      Challenge? @relation("ChallengeWinners", fields: [challenge_id], references: [id])
  

  @@map("user_challenges")
}
```

The point is: I don't want to create another field in the table UserChallenges named "winner_challenge_id", because it's the same value of challenge_id..
I have a user that is a UserChallenge and the same time a Winner.

The way it is now I get the error: _Error parsing attribute "@relation": The given constraint name `user_challenges_challenge_id_fkey` has to be unique in the following namespace: on model `UserChallenges` for primary key, indexes, unique constraints and foreign keys. Please provide a different name using the `map` argument._

Is there any way to do that, a virtual field or to put a alias, something like that?

Thanks for your support!

**Comment by rajesh40** [comment]
same here.....

**Comment by ArthurRF** [comment]
Any updates?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
